### DISCUSSION TITLE
I can t install nocodb on my raspberry [Oct 04]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#4014] [Discord Threads]
> 🙋‍♂️ **r4nchy**

**Comment by o1lab** [comment]
> 👨‍💻 **pranavxc**  
#### Can you share the error details ?  
#  
> 🙋‍♂️ **r4nchy**  
####   
![image0](https://cdn.discordapp.com/attachments/1024168208997023816/1024193514847862784/fTlLEKR2bz.png)  
#  
> 🙋‍♂️ **r4nchy**  
#### i have no idea using docker, so i went for manual installation  
#  
> 🙋‍♂️ **Acedia**  
#### Please share your installation steps with us, make sure executable has correct permission to run  
#  
> 🙋‍♂️ **Bergi9**  
#### i have tried it on my raspberry pi 3 model B (`Linux raspberrypi 5.15.28-v8+ #1532 SMP PREEMPT Tue Mar 15 13:19:15 GMT 2022 aarch64 GNU/Linux`) and it works. perhaps you didn't use a 64bit arm os.  
#  
> 🙋‍♂️ **Bergi9**  
#### I mean the execution of the executable works for me, but I got errors:   
#### ```   
#### Knex: run   
#### $ npm install sqlite3 --save   
#### Cannot find module '/snapshot/pkg-executable/node_modules/sqlite3/lib/binding/napi-v3-linux-arm64/node_sqlite3.node'   
#### Require stack:   
#### - /snapshot/pkg-executable/node_modules/sqlite3/lib/sqlite3-binding.js   
#### - /snapshot/pkg-executable/node_modules/sqlite3/lib/sqlite3.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/dialects/sqlite3/index.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/config-resolver.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/knex.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/index.js   
#### - /snapshot/pkg-executable/node_modules/knex/knex.js   
#### - /snapshot/pkg-executable/node_modules/nocodb/dist/bundle.js   
#### - /snapshot/pkg-executable/index.js   
#### 1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.   
#### Error: Cannot find module '/snapshot/pkg-executable/node_modules/sqlite3/lib/binding/napi-v3-linux-arm64/node_sqlite3.node'   
#### Require stack:   
#### - /snapshot/pkg-executable/node_modules/sqlite3/lib/sqlite3-binding.js   
#### - /snapshot/pkg-executable/node_modules/sqlite3/lib/sqlite3.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/dialects/sqlite3/index.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/config-resolver.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/knex.js   
#### - /snapshot/pkg-executable/node_modules/knex/lib/index.js   
#### - /snapshot/pkg-executable/node_modules/knex/knex.js   
#### - /snapshot/pkg-executable/node_modules/nocodb/dist/bundle.js   
#### - /snapshot/pkg-executable/index.js   
#### 1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.   
####     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)   
####     at Function._resolveFilename (pkg/prelude/bootstrap.js:1955:46)   
####     at Function.Module._load (node:internal/modules/cjs/loader:778:27)   
####     at Module.require (node:internal/modules/cjs/loader:1005:19)   
####     at Module.require (pkg/prelude/bootstrap.js:1855:31)   
####     at require (node:internal/modules/cjs/helpers:102:18)   
####     at Object.<anonymous> (/snapshot/pkg-executable/node_modules/sqlite3/lib/sqlite3-binding.js:4:15)   
####     at Module._compile (pkg/prelude/bootstrap.js:1930:22)   
####     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)   
####     at Module.load (node:internal/modules/cjs/loader:981:32)   
#### ```  
#  
> 👨‍💻 **pranavxc**  
#### On executable build we are copying the binaries for sqlite  for all platform and probably `napi-v3-linux-arm64/node_sqlite3.node` is missing.... we will include those files  
#  
> 👨‍💻 **pranavxc**  
#### hope that will resolve the issues  
#  
> 🙋‍♂️ **Bergi9**  
#### not for the OP, his issue is different than mine. it sounds like running a 64bit executable on a 32bit OS.  
#  
> 👨‍💻 **pranavxc**  
#### ok, he may be using docker or executable. In case of executable need to use the right build.  
#  
> 🙋‍♂️ **Bergi9**  
#### ```   
#### curl http://get.nocodb.com/linux-arm64 -o nocodb -L \   
####   && chmod +x nocodb \   
####   && ./nocodb   
#### ```   
#### that's not docker. And he said he didn't use docker.  
#  
> 👨‍💻 **pranavxc**  
#### Oh sorry my bad, I didn't seen that  
#  
> 🙋‍♂️ **pawan108**  
#### looks like RPI3 supports 64-bit apps when using non-raspbian OSes.  
#  
> 🙋‍♂️ **Bergi9**  
#### i use the raspbian OS, the aarch64 one which is 64bit  
#  
> 🙋‍♂️ **Bergi9**  
#### ah my fault, not the raspbian, the raspberry pi OS.  
#  
> 🙋‍♂️ **pawan108**  
#### it's the default one shipped with RPI3?  
#  
> 🙋‍♂️ **Bergi9**  
#### idk. i have it shipped with no sd card, so installed myself  
#  
> 🙋‍♂️ **pawan108**  
#### could you try npm install sqlite3 node?    
####    
#### say, npm install sqlite3  
#  
> 🙋‍♂️ **Bergi9**  
#### that's a packed executable, i have no nodejs/npm installed on my OS. so they didn't include the sqlite binary with the executable for this architecture.  
#  
> 🙋‍♂️ **pawan108**  
#### as <@682854952657616902> noted, perhaps the next release can pack the statically linked sqlite node library as part of the binary version  
#  
> 🙋‍♂️ **r4nchy**  
#### yes thats correct I am using armv7l raspberry os. Which is  I think a 32bit os  
#  
> 🙋‍♂️ **r4nchy**  
#### raspberry pi os 64bit was officially released this year, it would take time for people to shift to this new architecture as the 32bit was active for a decade and so many binaries of programs were made for armv7l.  
#  
> 🙋‍♂️ **r4nchy**  
#### 64bit os has advantage over using better RAM etc. So I even gave it a try by installing it on my raspberry pi 4b, but its so new that most previous projects won't even run.  
#  
> 👨‍💻 **pranavxc**  
#### can anyone share the static binaries for the architecture, it will be helpful  
#  
> 🙋‍♂️ **Bergi9**  
#### you mean by installing sqlite via npm on my raspberry then send it to you?  
#  
> 👨‍💻 **pranavxc**  
#### you can install nodejs and setup using our npx command   
####    
#### https://docs.nocodb.com/getting-started/installation#npx   
#### ```   
#### npx create-nocodb-app   
#### ```  
#  
> 👨‍💻 **pranavxc**  
#### if you can do the above then share the `node_modules/sqlite3/lib/binding/` folder content  
#  
> 🙋‍♂️ **Bergi9**  
#### sure, give me some time by installing nodejs first ^^  
#  
> 👨‍💻 **pranavxc**  
#### Sure take your time and thanks  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1024168208997023816)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
